common/src/types.rs

```diff
@@ -57,6 +58,26 @@ impl TimestampedPrice {
     pub fn as_fixed_point(&self) -> FixedPoint {
         FixedPoint::from_f64_round_down(self.price)
     }
+
+    /// Decimal correct a price for a given token pair
+    pub fn decimal_correct(
```

Comment:
nit: `get_decimal_corrected_price` would be a better name here. The current name seems to imply that we mutate the receiver